src/flood_fill.h

```diff
@@ -47,7 +47,11 @@ std::vector<Point> point_flood_fill_4_connected( const Point &starting_point,
         visited.emplace( current_point );
 
         if( predicate( current_point ) ) {
-            filled_points.emplace_back( current_point );
+            if constexpr( std::is_same_v<Container<Point>, std::vector<Point>> ) {
+                filled_points.emplace_back( current_point );
+            } else {
+                filled_points.emplace( current_point );
+            }
```

Comment:
`filled_points.emplace(filled_points.end(), current_point)` should work sensibly for all relevant containers.